nodejs/eks/cluster.ts

```diff
@@ -340,6 +340,20 @@ export function createCore(name: string, args: ClusterOptions, parent: pulumi.Co
         throw new Error("Setting nodeGroupOptions, and any set of singular node group option(s) on the cluster, is mutually exclusive. Choose a single approach.");
     }
 
+    // Check if the cluster name provided or autoname used starts with 'eks-'.
+    // The cluster name is used for the fargate profile name, and we must
+    // ensure `eks-` is not used in either naming option, as AWS considers it
+    // a reserved prefix and will fail the fargate profile creation.
+    if (args.fargate) {
+        const nameToCheck = args.name || name.concat("-");
+        const profileNameRegex = new RegExp("^" + "eks-", "i"); // starts with (^) 'eks-', (i)gnore casing
+        pulumi.output(nameToCheck).apply(n => {
+            if (profileNameRegex.test(n)) {
+                throw new Error(`The fargate profile name uses the cluster name, and the name provided will use the reserved AWS prefix that is not allowed: 'eks-'. Choose an alternate name`);
+            }
+        });
```

Comment:
Does this approach of asyncronously throwing an error that isn't "handled" anywhere actually work?  I wouldn't have thought so.  This code will continue on executing the body of this function, and only once done with all the synchronous work of constructing these resources will it come back to asynchronously execute this callback.  And then that promise/output will fail with an error, but nothing is listening to it (it's not an input to anything else) - so I think it will just get ignored.  It may finally get picked up by Node's unhandled promise rejection handler a couple turns later - but I suspect that (a) the experience won't be good and (b) other resources may well get created ahead of this.  

Relatedly - can't we just construct a name that will actually work instead of trying to error?  That is can't we just change this line:

```
fargateProfile = new aws.eks.FargateProfile(`${name}-fargateProfile`, {
```

To check if `name` is `eks` and if so use an `_` instead of `-` or something like that (or just drop the `-`)?